scratch-js/Sprite.mjs

```diff
@@ -187,24 +187,15 @@ export class Sprite extends SpriteBase {
   }
 
   createClone() {
-    const parent = this.parent || this;
     const clone = Object.assign(
-      Object.create(Object.getPrototypeOf(parent)),
-      parent
+      Object.create(Object.getPrototypeOf(this)),
+      this
     );
 
-    clone._project = parent._project;
-
-    clone.triggers = parent.triggers.map(
-      trigger =>
-        new Trigger(
-          trigger.trigger,
-          trigger.options,
-          trigger._script.bind(clone)
-        )
-    );
-    clone.costumes = parent.costumes;
-    clone._vars = Object.assign({}, parent._vars);
+    clone._project = this._project;
+    clone.triggers = this.triggers;
```

Comment:
Is it alright that we're not cloning the triggers anymore?

Reply:
I believe so. The method attached to the trigger is specifically **not** bound to a particular instance (which is the opposite of how it was before clones were introduced). The method being called is bound with the correct value of `this` *when the trigger fires*, which means that copying the triggers array directly from another sprite shouldn't cause any problems. In theory, anyway.

Reply:
It looks like the trigger's internal state (e.g. generator state) will be shared between the parent and clone, though. If you call `start()` on the parent trigger, won't the child trigger start over from the beginning too?

Reply:
Oh! I think you're right. Will change.

Reply:
How does the new code look?

Reply:
👍